pyne/mesh.py

```diff
@@ -704,9 +705,18 @@ def __init__(self, mesh=None, structured=False,
             stored in self.dims.
 
         """
+        # obtain param names and default values
+        args, _, _, defaults = inspect.getargspec(Mesh.__init__)
```

Comment:
Great move using `inspect` to get the function signature. I'll just recommend a few updates to reduce the number of lines in our check: 

```python

args, _, _, defaults = inspect.getargspec(Mesh.__init__)
# join args into tuple, eval will recognize the comma as a delimiter and create a tuple of args
args = eval(",".join(args))
allDefault = args == defaults
```

Reply:
Another option, which maybe is more readable would be: 

```python
args = tuple([eval(arg) for arg in args[1:]])
```